### PR TITLE
add Dynamic Libraries info to Contributor guide

### PR DESCRIPTION
**Pull request by PeterTurcan**
@alandefreitas added a section on dynamic libs to the Design Guide - based on your issue. Not sure if I have covered all that is needed.

### CODE REVIEW

File: contributor-guide/modules/ROOT/pages/design-guide/design-guide.adoc [hunk 1]
```diff
@@ -100,6 +100,20 @@ Make sure your code compiles in the presence of the `min()` and `max()` macros.
...
 
+== Support for Dynamic Libraries
+
+A dynamic library is a compiled collection of binary code and data that can be loaded and linked into a program at runtime, allowing shared use of library functions across multiple applications without the need for recompilation. Unlike static libraries, which are linked at compile time and become part of the resulting executable, dynamic libraries remain separate files and are often employed to save memory, reduce redundancy, and facilitate modular application design.
+
+This concept is known by different names on various platforms: shared objects (.so) on UNIX systems, dynamic link libraries (.dll) on Windows, and dynamic shared libraries (.dylib) on macOS.
+
+If your plan for your library is to enable its use as a dynamic library, you need to implement two sets of macros:
```
Comment (alandefreitas) [inline contributor-guide/modules/ROOT/pages/design-guide/design-guide.adoc:109]: I'm not sure if providing the functionality is optional. I'm not saying it's not. I really don't know.

> you need to implement two sets of macros:

The macros are implemented. They just need to use them.

File: contributor-guide/modules/ROOT/pages/design-guide/design-guide.adoc [hunk 2]
```diff
@@ -100,6 +100,20 @@ Make sure your code compiles in the presence of the `min()` and `max()` macros.
...
+
+. https://www.boost.org/doc/libs/master/libs/config/doc/html/boost_config/boost_macro_reference.html#boost_config.boost_macro_reference.macros_for_libraries_with_separate_source_code.macros_controlling_shared_library_symbol_visibility[Macros controlling shared library symbol visibility]
+
+. https://www.boost.org/doc/libs/master/libs/config/doc/html/boost_config/boost_macro_reference.html#boost_config.boost_macro_reference.macros_for_libraries_with_separate_source_code.automatic_library_selection[Automatic library selection]
+
+As untested macros are error-prone, it is important that they are thoroughly tested on a full range of compilers. For a full list of compilers, refer to  xref:user-guide:ROOT:header-organization-compilation.adoc#toolset[Toolsets].
```
Comment (alandefreitas) [inline contributor-guide/modules/ROOT/pages/design-guide/design-guide.adoc:115]: > As untested macros are error-prone, it is important that they are thoroughly tested on a full range of compilers.

I don't understand this sentence.

**Comment by alandefreitas** [comment]
> based on your issue

You have to link PRs to issues. Via a `fix #xx` in the commit message footer and in 

image

The footer should work but it might fail if the PR is being amended.

**Comment by PeterTurcan** [comment]
Closing pending more research

**Pull request closed without merging**